db/error_handler.cc

```diff
@@ -406,10 +406,7 @@ const Status& ErrorHandler::SetBGError(const IOStatus& bg_io_err,
     // it can directly overwrite any existing bg_error_.
     bool auto_recovery = false;
     Status bg_err(new_bg_io_err, Status::Severity::kUnrecoverableError);
-    bg_error_ = bg_err;
-    if (recovery_in_prog_ && recovery_error_.ok()) {
-      recovery_error_ = bg_err;
-    }
+    CheckAndSetRecoveryAndBGError(bg_err);
```

Comment:
Do we know why the original code wasn't checking bg_error_ severity before setting it?

Reply:
The io_error is classified as"Status::Severity::kUnrecoverableError", which is the worst one. It is safe to directly override bg_error_.